pkg/api/v20210131preview/openshiftcluster_validatestatic_test.go

```diff
@@ -462,6 +463,63 @@ func TestOpenShiftClusterStaticValidateNetworkProfile(t *testing.T) {
 	runTests(t, testModeUpdate, tests)
 }
 
+func TestOpenShiftClusterStaticValidateNetworkProfileType(t *testing.T) {
+	createtests := []*validateTest{
+		{
+			name: "valid",
+		},
+		{
+			name: "networkProvider",
+			modify: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.SDNProvider = "InvalidOption"
+			},
+			wantErr: "400: InvalidParameter: properties.networkProfile.sdnProvider: The provided SDNProvider 'InvalidOption' is invalid.",
+		},
+		{
+			name: "networkProvider",
+			modify: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.SDNProvider = "OpenShiftSDN"
+			},
+		},
+		{
+			name: "networkProvider",
+			modify: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.SDNProvider = "OVNKubernetes"
+			},
+		},
+	}
+
+	updatetests := []*validateTest{
+		{
+			name: "valid",
+		},
+		{
+			name: "networkProvider",
+			modify: func(oc *OpenShiftCluster) {
+				oc.Properties.NetworkProfile.SDNProvider = "InvalidOption"
+			},
+			wantErr: "400: InvalidParameter: properties.networkProfile.sdnProvider: The provided SDNProvider 'InvalidOption' is invalid.",
```

Comment:
`"Changing property '*' is not allowed"` tests need to go `TestOpenShiftClusterStaticValidateDelta` test which tests immutability of the fields.

Reply:
Moved.